### DISCUSSION TITLE
D-ITG-2.8.1-r1023-Win

### DISCUSSION THREAD

**Discussion by tabeshforoughi** [esnet/iperf#1449] [Q&A] [answered]
Please provide a command for measuring the RTT of TCP and UDP with IPV4 and IPV6 using D-ITG-2.8.1-r1023-Win.
Please mention how it can be used for IPV4 or IPV6.
Many Thanks

**Comment by davidBar-On** [comment, marked as answer]
@tabeshforoughi, this is an iperf3 discussion list, not D-ITG, so I am not sure you will be able to get any help related to D-ITG here.  You may try to use the D-ITG mailing list as mentioned in the [D-ITG Github site](https://github.com/jbucar/ditg).

A common tool for measuring RTT is [ping](https://linux.die.net/man/8/ping).  However, it measures only the RTT between the client and server machines, and does not include the UDP/TCP overhead.

With iperf3, use `-J` option.  The output includes `max/min/mean_rtt` values for the test, which can be UDP or TCP, IPv4 or IPv6 (use `iperf3 -h` for help on how to set these parameters).

Per the [D-ITG exampe here](http://sdnopenflow.blogspot.com/2015/05/using-of-d-itg-traffic-generator-in.html), it seems that D-ITG does not compute the RTT.  You may log the traffic and use Wireshark for the analysis.

**Reply by tabeshforoughi** [reply]
Thank you, Mr. Dave.
